src/Microsoft.ML.Legacy/Runtime/Internal/Tools/CSharpApiGenerator.cs

```diff
@@ -115,10 +116,10 @@ private void GenerateInputOutput(IndentingTextWriter writer, ComponentCatalog.En
             GenerateInput(writer, entryPointInfo, catalog);
             writer.Outdent();
             writer.WriteLine("}");
-            writer.WriteLine();
+            writer.WriteLineNoTabs();
```

Comment:
Why is this change necessary?  here and below

Reply:
Yes, it's necessary.

`writer.WriteLine();` will output tabs